qemu/tests/spapr_vty_multi_backends.py

```diff
@@ -0,0 +1,76 @@
+import re
+import os
+
+from virttest import env_process
+from virttest import error_context
+from virttest import remote
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Verify the login guest with multi backends of pci-serial (RHEL only):
+    1) Boot guest with multi pci-serial with backend
+    2) Modify the kernel cfg file to specify the backend
+    3) for pty and file backend:
+      3.1) open and close chardev
+    4) for unix_socket and tcp_socket
+      4.1) Login guest.
+      4.2) move, create files inside guest
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def check_guest():
+        session.cmd(params['create_file_and_copy'])
+
+    for serial_id in list(filter(lambda x: x.startswith('vs') is True,
+                                 params.objects('serials'))):
+        if serial_id != "vs1":
+            hvc_id = int(serial_id.split('vs', 1)[1]) - 1
+            params['kernel_extra_params_add'] = "console=hvc%s,115200" % hvc_id
+        prompt = params.get("shell_prompt")
+        params['start_vm'] = 'yes'
+        env_process.preprocess(test, params, env)
```

Comment:
preprocess multiple times is not a good idea. You can preprocess once and use `vm.create` after that.

Reply:
Okay, let me experiment. Update later.